runtime/src/block_min_prioritization_fee_cache.rs

```diff
@@ -0,0 +1,595 @@
+                "block_min_prioritization_fee_counters",
+                ("slot", slot as i64, i64),
+                (
+                    "fail_finalize_block_not_found",
+                    self.fail_finalize_block_not_found as i64,
+                    i64
+                ),
+            );
+            self.successful_transaction_update_count = 0;
+            self.fail_get_transaction_priority_details_count = 0;
+            self.fail_get_transaction_account_locks_count = 0;
+            self.fail_finalize_block_not_found = 0;
+        }
+    }
+}
+
+impl BlockMinPrioritizationFeeCacheUpdate for BlockMinPrioritizationFeeCache {
+    /// Update block's min prioritization fee with `txs`,
+    /// Returns updated min prioritization fee for `slot`
+    fn update_transactions<'a>(
+        &mut self,
+        slot: Slot,
+        txs: impl Iterator<Item = &'a SanitizedTransaction>,
+    ) -> Option<u64> {
+        let updated_block_min_fee: Option<u64>;
+
+        let mut successful_transaction_update_count: u64 = 0;
+        let mut fail_get_transaction_priority_details_count: u64 = 0;
+        let mut fail_get_transaction_account_locks_count: u64 = 0;
+
+        {
+            let block = self.get_or_add_mut_block_min_prioritization_fee(&slot);
+
+            for sanitized_tx in txs {
+                match block.update_for_transaction(sanitized_tx) {
+                    Err(BlockMinPrioritizationFeeError::FailGetTransactionPriorityDetails) => {
+                        saturating_add_assign!(fail_get_transaction_priority_details_count, 1)
+                    }
+                    Err(BlockMinPrioritizationFeeError::FailGetTransactionAccountLocks) => {
+                        saturating_add_assign!(fail_get_transaction_account_locks_count, 1)
+                    }
+                    _ => saturating_add_assign!(successful_transaction_update_count, 1),
+                }
+            }
+
+            updated_block_min_fee = block.get_block_fee();
+        }
+
+        saturating_add_assign!(
+            self.successful_transaction_update_count,
+            successful_transaction_update_count
+        );
+        saturating_add_assign!(
+            self.fail_get_transaction_priority_details_count,
+            fail_get_transaction_priority_details_count
+        );
+        saturating_add_assign!(
+            self.fail_get_transaction_account_locks_count,
+            fail_get_transaction_account_locks_count
+        );
+
+        updated_block_min_fee
+    }
+
+    /// bank is completely replayed from blockstore, prune irrelevant accounts to save space,
+    /// its fee stats can be made available to queries
+    fn finalize_block(&mut self, slot: Slot) {
+        if let Some(block) = self.get_mut_block_min_prioritization_fee(&slot) {
+            block.prune_irrelevant_accounts();
+            let _ = block.mark_block_completed();
+        } else {
+            saturating_add_assign!(self.fail_finalize_block_not_found, 1);
```

Comment:
Exactly, hope it never hit and that counter in metric stays zero. I just thought in case something bad/weird ever happen, we have something to look at. But it might be overdone.

Reply:
Think again, you are right that no need to report `0` if this isn't likely to be hit. Removed the reporting, replaced with  `warn` as least for initial testing